app/Helpers/Bank/Nordigen/Nordigen.php

```diff
@@ -61,18 +61,32 @@ public function getInstitutions()
     }
 
     // requisition-section
-    public function createRequisition(string $redirect, string $institutionId, string $reference, string $userLanguage)
+    public function createRequisition(string $redirect, string $institutionId, string $reference, string $userLanguage, int $maxHistoricalDays = null, int $accessValidForDays = null)
```

Comment:
@dshoreman I have concluded this today, so you can work based on that change to reduce total work of merging on your side.
I have update the Nordigen function createRequisition to be able to pass maxHistoricalDays and accessValidForDays, so you can accept the changes to the Nordigen class in total and "just" implement this new function to your changes required for the UI.

Therefore the ui should pass the variables while sending the request f.ex. as query parameters and the view should handle the errors of the NordigenClass accordingly.

If you have any questions, feeld free to reach out to me :)

Reply:
I'm not sure this is all necessary, a lot of the handling is already done in #10425 so this just confuses things more for the rebase, especially with parts copied from that PR too.

Still finding my way through comments but I'll try to get on Slack at some point. It may be easier if I rebase first (on current v5-dev before this is merged) then we add the access valid option after.

Reply:
I think this PR is indeed necessary as of this puts the logic inside of the NordigenClass instead of the Controller to handle the auth flow and leverages the already existing aproach from PR #10410

@dshoreman I copied the 1 method to get the endUserAgreement by id and one line of the handling of the past days within the controller, new within this PR is the reusability of outstanding agreements and the loading mechanism of the default values. Your refactor is still relevant and should be handled within your PR and is more focused on the view.

If you need any help doing the rebase, i can assist if you like :)